### PR TITLE
Manual backport 1.15.x of Disable remote proxy patching except AWS Lambda

### PR DESCRIPTION
**Pull request by zalimeni**
backport of commit ba18381ab20b2e131d1dd4cfc642bfadf6b78b26

### Description

Manual backport of https://github.com/hashicorp/consul/pull/17415.

Resolves automated backport PR https://github.com/hashicorp/consul/pull/17428.

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -2,7 +2,7 @@ module github.com/hashicorp/consul/api
 
 go 1.19
 
-//replace github.com/hashicorp/consul/sdk => ../sdk
+replace github.com/hashicorp/consul/sdk => ../sdk
```
Comment (zalimeni) [inline api/go.mod:5]: For consistency, I also updated this one even though I didn't change `sdk`; see comment [here](https://github.com/hashicorp/consul/pull/17432/files#r1202563919)

**Review by zalimeni — COMMENTED** [review]
I've noted the only differences between `main` and `1.15` above; beyond moved references and fatter tests, there are no functional changes. Happy to discuss any specifics if helpful, though.

**Review by hashi-derek — APPROVED** [review]
Approving the `go.mod` changes for this backport. I'm assuming that the original enterprise review was correct on content.

**Comment by zalimeni** [comment]
Rebased for merge

**Pull request merged**